tests/test_auto_layouting.py

```diff
@@ -97,7 +92,8 @@ def select(self, expression=None):
         assume(elements)
         return sampled_from(elements)
 
-    def relations(self, diagram):
+    @staticmethod
```

Comment:
I know IDEs often provide guidance that a method could be static. I would avoid this unless you want to directly call AutoLayouting.relations(). You could also move it from a method to a function outside of this class.

Reply:
thanks @danyeaw, I indeed followed my IDE's suggestion on this without too much thought. I agree with your suggestion to move it outside the class, which is now implemented.